datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -162,12 +173,30 @@ func (f *ChainDataFetcher) resLoop() {
 				logger.Error("db insertion is failed", "blockNumber", res.blockNumber, "reqType", res.reqType, "err", res.err)
 				// TODO-ChainDataFetcher add retry logic when data insertion is failed
 			} else {
-				f.updateCheckpoint(res.blockNumber.Uint64())
+				if err := f.updateCheckpoint(res.blockNumber.Int64()); err != nil {
```

Comment:
Don't we need to retry if updating checkpoint fails?

Reply:
I added comment for retry on 5c157a079db7ed06950b5cc091ff12cac9c8ee1e. I will handle this with the above retry handling TODO.